### PR TITLE
Counter rtc legacy api

### PR DESCRIPTION
**Pull request by nordic-krch**
Commit to look at: https://github.com/zephyrproject-rtos/zephyr/commit/de313328c09c585c8427c77fb0a07f6c4dd8adab

I would like to get an opinion on how to do deprecation. The problem is that legacy function `counter_set_alarm` exists in new API but for different purpose.
Legacy `counter_set_alarm` maps almost directly to new `counter_set_wrap`.

In this PR i've renamed new set alarm to `counter_set_ch_alarm` as for channel alarm. That is because it's setting alarm for given channel. What's your opinion? Rename this function and have legacy one as deprecated (like in this draft commit https://github.com/zephyrproject-rtos/zephyr/commit/de313328c09c585c8427c77fb0a07f6c4dd8adab) or leave it and has no backward compatibility?

### CODE REVIEW

**Comment by carlescufi** [comment]
@nordic-krch should this be targetting `topic-counters` instead of `master`?

**Comment by nordic-krch** [comment]
@anangl @MaureenHelm @galak could you take a look at this. This should go into topic-counters before any rtc counter implementions go in so that they can add function that is needed by compatibility layer (`counter_get_user_data()`)

**Review by nashif — APPROVED** [review]

File: include/rtc.h [hunk 1]
```diff
@@ -57,49 +58,64 @@ struct rtc_driver_api {
...
+__deprecated __syscall int rtc_set_alarm(struct device *dev,
+					 const u32_t alarm_val);
 
 static inline int _impl_rtc_set_alarm(struct device *dev,
 				      const u32_t alarm_val)
 {
-	const struct rtc_driver_api *api = dev->driver_api;
-
-	return api->set_alarm(dev, alarm_val);
+	return counter_set_wrap(dev, alarm_val, rtc_counter_wrap_callback,
```
Comment (MaureenHelm) [inline include/rtc.h]: Why isn't this `counter_set_alarm`?

File: include/counter.h [hunk 2]
```diff
@@ -389,6 +394,33 @@ static inline u32_t _impl_counter_get_max_relative_alarm(struct device *dev)
 }
 #endif
 
+__deprecated static inline int counter_set_alarm(struct device *dev,
+						 counter_callback_t callback,
+						 u32_t count, void *user_data)
+{
+	return counter_set_wrap(dev, count, callback, user_data);
```
Comment (MaureenHelm) [inline include/counter.h]: Why isn't this `counter_set_alarm`?

**Pull request merged**